python/ray/actor.py

```diff
@@ -187,8 +188,21 @@ def export_actor(actor_id, Class, actor_method_names, num_cpus, num_gpus,
   local_scheduler_id, gpu_ids = select_local_scheduler(local_schedulers,
                                                        num_gpus, worker)
 
-  worker.redis_client.publish("actor_notifications",
-                              actor_id.id() + local_scheduler_id)
+  # Publish a message to notify the relevant local scheduler that it needs to
+  # create an actor. If necessary, retry until the local scheduler actually
+  # gets the message.
+  num_retries = 10
+  for _ in range(num_retries):
+    num_recipients = worker.redis_client.publish(
+        "actor_notifications", actor_id.id() + local_scheduler_id)
+    if num_recipients == 1:
```

Comment:
This doesn't quite work because all of the local schedulers in the cluster will receive this publish message.

We could separate this into two publish message:
- One targeted at the local scheduler that is supposed to create the actor.
- One targeted at all local schedulers to notify them of the existence of a new actor.